src/api/cvc4cpp.cpp

```diff
@@ -2010,31 +1917,72 @@ Term Solver::mkConst(Kind kind, uint32_t arg1, uint32_t arg2, Term arg3) const
 
 Term Solver::mkVar(const std::string& symbol, Sort sort) const
 {
-  // CHECK: sort exists?
   return d_exprMgr->mkVar(symbol, *sort.d_type);
 }
 
 Term Solver::mkVar(Sort sort) const
 {
-  // CHECK: sort exists?
   return d_exprMgr->mkVar(*sort.d_type);
 }
 
 Term Solver::mkBoundVar(const std::string& symbol, Sort sort) const
 {
-  // CHECK: sort exists?
   return d_exprMgr->mkBoundVar(symbol, *sort.d_type);
 }
 
 Term Solver::mkBoundVar(Sort sort) const
 {
-  // CHECK: sort exists?
   return d_exprMgr->mkBoundVar(*sort.d_type);
 }
 
 /* Create terms                                                               */
 /* -------------------------------------------------------------------------- */
 
+void Solver::check_mk_term(Kind kind, uint32_t nchildren) const
+{
+  PrettyCheckArgument(isDefinedKind(kind),
+                      kind,
+                      "Invalid kind '%s'",
+                      kindToString(kind).c_str());
+  const CVC4::kind::MetaKind mk = kind::metaKindOf(extToIntKind(kind));
+  PrettyCheckArgument(
+      mk == kind::metakind::PARAMETERIZED || mk == kind::metakind::OPERATOR,
```

Comment:
Shouldn't this be `mk != kind::metakind::PARAMETERIZED && mk != kind::metakind::OPERATOR`?

Reply:
No. See https://github.com/CVC4/CVC4/blob/master/src/expr/expr_manager_template.cpp#L172.

Reply:
Right, that makes sense. I got confused by the original comment (because of the `else` part): https://github.com/CVC4/CVC4/pull/2455/files#diff-d4eff0da8e72b4f2d088f43ce4ecf899L2069 .